### ISSUE TITLE
Cart rule ignores condition with categories

### ISSUE THREAD

**Issue by kerlama** [magento/magento2#18767]
We have problem on latest 2.2.6 and also on 2.1.15 version on excluding categories which contains only bundle products when creating coupon.

### Preconditions (*)
1. Magento 2.2.6 and 2.1.15
2. PHP 7.0.30

### Steps to reproduce (*)
1. Create category called Bundle, for example
2. Create 2 simple products
3. Create bundle product, assign one simple product to it and assign it to "Bundle" category
4. Create cart rule with these conditions:
If **ALL** of these conditions are **TRUE**:
Subtotal **equals or greater than 50**
If an item is **FOUND** in the cart with **ALL** of these conditions true:
Category **is not one of 2**

Note that category id 2 is "Bundle" category. And the price of simple products is $20 for example, and bundle is $40.

5. Set actions to apply percent of product price discount to 10 and these rule conditions:
If **ALL** of these conditions are **TRUE**:
Category **is not one of 2**

6. Save rule
7. Go to store, add simple product which is not part of the bundle and bundle. Subtotal should be $60 Which is fine
8. Apply coupon code
9. Coupon is applied and you got discount, which is wrong

### Expected result (*)
1. Magento should throw message that coupon is not valid because cart contain item which belongs to category "Bundle".

### Actual result (*)
1. Code is applied and you got discount on both items

**Comment by magento-engcom-team** [comment]
Hi @kerlama. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@kerlama do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by kerlama** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @kerlama. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @kerlama, here is your [Magento instance](http://ec2-34-228-235-121.compute-1.amazonaws.com/i-18767-2-3-develop/).
Admin access: [http://ec2-34-228-235-121.compute-1.amazonaws.com/i-18767-2-3-develop//admin](http://ec2-34-228-235-121.compute-1.amazonaws.com/i-18767-2-3-develop//admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by kerlama** [comment]
@magento-engcom-team demo is not working as expected, I could not add items to cart, getting session expired message or invalid form key.

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-pb. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by ghost** [comment]
@kerlama, thank you for your report.
Please make your steps to reproduce more consistent and clear.

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-pb. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by ghost** [comment]
@kerlama, we are closing this issue due to inactivity. If you'd like to update it, please reopen the issue.
